File: react-native-pytorch-core/ios/Canvas/ImageDataModule.m
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

#import <Foundation/Foundation.h>
#import <React/RCTBridgeModule.h>
#import <React/RCTConvert.h>

@interface RCT_EXTERN_REMAP_MODULE(PyTorchCoreImageDataModule, ImageDataModule, NSObject)

RCT_EXTERN_METHOD(release:(nonnull NSDictionary)imageDataRef resolver:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject);

@end

File: react-native-pytorch-core/ios/Canvas/ImageDataModule.swift
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import Foundation

@objc(ImageDataModule)
public class ImageDataModule: NSObject {

    @objc
    public func getName() -> String {
        return "PyTorchCoreImageDataModule"
    }

    @objc(release:resolver:rejecter:)
    public func release(_ imageDataRef: NSDictionary,
                        resolver resolve: RCTPromiseResolveBlock,
                        rejecter reject: RCTPromiseRejectBlock) {
        do {
            if let imageDataRef = imageDataRef as? [ String: String] {
                try JSContext.release(jsRef: imageDataRef)
            }
            resolve(nil)
        } catch {
            reject(RCTErrorUnspecified, "Invalid image data reference in release: \(error)", error)
        }
    }
}
